### PR TITLE
TST add tests for full dep update

### LINKED ISSUE #1611: Metadata getting appended to version in bot updates
```toml
importlib-metadata = {version = ">=1.6.0", python = "<3.8"} 
```
Got translated as:

image
* https://github.com/conda-forge/dunamai-feedstock/pull/15/commits/0e8c0e6310c8009c02ed7c089d3a61dd1917ffff
* https://github.com/mtkennerly/dunamai/blob/v1.16.0/pyproject.toml#L22

----

```toml
poetry = { version = "^1.2.0", optional = true }
```
Got translated as:

image
* https://github.com/conda-forge/poetry-dynamic-versioning-feedstock/pull/14/commits/04cb99bfbb9b538a7bdb77a41b550355b0f58860
* https://github.com/mtkennerly/poetry-dynamic-versioning/blob/v0.21.4/pyproject.toml#L29

### LINKED ISSUE #1617: Dep analysis appears broken; reports all meta.yml deps as missing on multiple feedstocks
The dependency analysis on autotick PRs appears to be broken, per my observations on multiple new autotick PRs in the past 24 hours, as it is suddenly listing all the deps listed in the `meta.yml` as missing, despite them still being present in the source code with no major changes. Moreover, it is also apparently not listing "missing" deps in the `meta.yml` that were imported (in non-package code) that were detected before.

I've observed this on both autotick bot PRs I reviewed and merged today, conda-forge/pyroma-feedstock#18 and conda-forge/praw-feedstock#24 . The URLs for those runs are https://github.com/regro/cf-scripts/actions/runs/4270734559 and https://github.com/regro/cf-scripts/actions/runs/4273594672 respectively.

As I haven't reviewed/merged a similar PR in at least several weeks, I can't confirm when this started, but it seems to be more than a one-off. Could you investigate? Thanks!

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds tests and possibly fixes for dep updates.

Closes #1617

### CODE REVIEW

**Comment by beckermr** [comment]
@ericdill @ocefpaf So this one is super weird. The dependency hints in the bot from depfinder appear to be broken in CI but work just fine on my local machine. I've dug a lot and the bug appears to be the data coming from depfinder. I've diff'ed the envs and not seen anything suspicious. 

Any ideas?

env diff:

```diff
1c1
< # packages in environment at /Users/beckermr/mambaforge/envs/bot-dev:
---
> # packages in environment at /Users/runner/miniconda3/envs/test:
4,6c4
< _ipython_minor_entry_point 8.7.0                h8cf3c4a_0    conda-forge
< alabaster                 0.7.12                     py_0    conda-forge
< anyio                     3.6.2              pyhd8ed1ab_0    conda-forge
---
> alabaster                 0.7.13             pyhd8ed1ab_0    conda-forge
8,11d5
< appnope                   0.1.3              pyhd8ed1ab_0    conda-forge
< argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
< argon2-cffi-bindings      21.2.0           py39ha30fb19_3    conda-forge
< asttokens                 2.2.1              pyhd8ed1ab_0    conda-forge
15d8
< backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
19c12
< beautifulsoup4            4.11.1             pyha770c72_0    conda-forge
---
> beautifulsoup4            4.11.2             pyha770c72_0    conda-forge
21d13
< bleach                    5.0.1              pyhd8ed1ab_0    conda-forge
41c33
< cloudpickle               2.2.0              pyhd8ed1ab_0    conda-forge
---
> cloudpickle               2.2.1              pyhd8ed1ab_0    conda-forge
44d35
< comm                      0.1.2              pyhd8ed1ab_0    conda-forge
46,47c37,38
< conda-build               3.23.3           py39h6e9494a_0    conda-forge
< conda-forge-feedstock-check-solvable 0.1.0                     dev_0    <develop>
---
> conda-build               3.23.3           py39h6e9494a_1    conda-forge
> conda-forge-feedstock-check-solvable 0.1.0              pyhd8ed1ab_0    conda-forge
49a41
> conda-libmamba-solver     23.1.0             pyhd8ed1ab_0    conda-forge
51d42
< conda-package-streaming   0.7.0              pyhd8ed1ab_1    conda-forge
56,57c47,48
< coverage                  7.0.1            py39ha30fb19_0    conda-forge
< cryptography              38.0.4           py39hbeae22c_0    conda-forge
---
> coverage                  7.2.1            py39ha30fb19_0    conda-forge
> cryptography              39.0.1           py39hbeae22c_0    conda-forge
61,63d51
< debugpy                   1.6.6            py39h7a8716b_0    conda-forge
< decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
< defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
70d57
< entrypoints               0.4                pyhd8ed1ab_0    conda-forge
73d59
< executing                 1.2.0              pyhd8ed1ab_0    conda-forge
76,77c62
< filelock                  3.8.2              pyhd8ed1ab_0    conda-forge
< flake8                    6.0.0              pyhd8ed1ab_0    conda-forge
---
> filelock                  3.9.0              pyhd8ed1ab_0    conda-forge
79d63
< flit-core                 3.8.0              pyhd8ed1ab_0    conda-forge
85c69
< fontconfig                2.14.1               h5bb23bf_0    conda-forge
---
> fontconfig                2.14.2               h5bb23bf_0    conda-forge
91,92c75,76
< fsspec                    2022.11.0          pyhd8ed1ab_0    conda-forge
< gdk-pixbuf                2.42.8               h710d9fd_2    conda-forge
---
> fsspec                    2023.1.0           pyhd8ed1ab_0    conda-forge
> gdk-pixbuf                2.42.10              he00a9fc_0    conda-forge
95c79
< git                       2.39.0          pl5321h0195497_0    conda-forge
---
> git                       2.39.2          pl5321h0195497_0    conda-forge
98,100c82
< gitpython                 3.1.29             pyhd8ed1ab_0    conda-forge
< glib                      2.74.1               hbc0c0cd_1    conda-forge
< glib-tools                2.74.1               hbc0c0cd_1    conda-forge
---
> gitpython                 3.1.31             pyhd8ed1ab_0    conda-forge
104c86
< graphviz                  6.0.2                hc51f7b9_0    conda-forge
---
> graphviz                  7.1.0                hc51f7b9_0    conda-forge
106,107d87
< gst-plugins-base          1.21.3               h37e1711_1    conda-forge
< gstreamer                 1.21.3               h1d18e73_1    conda-forge
113c93
< identify                  2.5.11             pyhd8ed1ab_0    conda-forge
---
> identify                  2.5.18             pyhd8ed1ab_0    conda-forge
117,122c97
< importlib_resources       5.10.1             pyhd8ed1ab_1    conda-forge
< iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
< ipykernel                 6.21.2             pyh736e0ef_0    conda-forge
< ipython                   8.7.0              pyhd1c38e8_0    conda-forge
< ipython_genutils          0.2.0                      py_1    conda-forge
< ipywidgets                8.0.4              pyhd8ed1ab_0    conda-forge
---
> iniconfig                 2.0.0              pyhd8ed1ab_0    conda-forge
124d98
< jedi                      0.18.2             pyhd8ed1ab_0    conda-forge
127,137c101
< jpeg                      9e                   hac89ed1_2    conda-forge
< jsonschema                4.17.3             pyhd8ed1ab_0    conda-forge
< jupyter                   1.0.0            py39h6e9494a_8    conda-forge
< jupyter_client            7.4.8              pyhd8ed1ab_0    conda-forge
< jupyter_console           6.4.4              pyhd8ed1ab_0    conda-forge
< jupyter_core              5.1.0            py39h6e9494a_0    conda-forge
< jupyter_events            0.5.0              pyhd8ed1ab_1    conda-forge
< jupyter_server            2.0.5              pyhd8ed1ab_1    conda-forge
< jupyter_server_terminals  0.4.3              pyhd8ed1ab_0    conda-forge
< jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
< jupyterlab_widgets        3.0.5              pyhd8ed1ab_0    conda-forge
---
> jpeg                      9e                   hb7f2c08_3    conda-forge
146d109
< libclang                  13.0.1          default_he082bbe_0    conda-forge
148,149c111,112
< libcxx                    14.0.6               hccf4f1f_0    conda-forge
< libdeflate                1.14                 hb7f2c08_0    conda-forge
---
> libcxx                    15.0.7               h71dddab_0    conda-forge
> libdeflate                1.17                 hac1461d_0    conda-forge
154,155c117,118
< libgfortran               5.0.0           11_3_0_h97931a8_27    conda-forge
< libgfortran5              11.3.0              h082f757_27    conda-forge
---
> libgfortran               5.0.0           10_4_0_h97931a8_28    conda-forge
> libgfortran5              11.3.0              h082f757_28    conda-forge
160d122
< libllvm13                 13.0.1               h64f94b2_2    conda-forge
165d126
< libogg                    1.3.4                h35c211d_1    conda-forge
167d127
< libopus                   1.3.1                hc929b4f_1    conda-forge
169d128
< libpq                     15.1                 h3640bf0_2    conda-forge
175,177c134,135
< libtiff                   4.5.0                h6268bbc_0    conda-forge
< libtool                   2.4.6             he49afe7_1008    conda-forge
< libvorbis                 1.3.7                h046ec9c_0    conda-forge
---
> libtiff                   4.5.0                hee9004a_2    conda-forge
> libtool                   2.4.7                hf0c8a7f_0    conda-forge
183c141
< llvm-openmp               15.0.6               h61d9ccf_0    conda-forge
---
> llvm-openmp               15.0.7               h61d9ccf_0    conda-forge
186c144
< lz4-c                     1.9.3                he49afe7_1    conda-forge
---
> lz4-c                     1.9.4                hf0c8a7f_0    conda-forge
189,192c147
< markupsafe                2.1.1            py39ha30fb19_2    conda-forge
< matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
< mccabe                    0.7.0              pyhd8ed1ab_0    conda-forge
< mistune                   2.0.4              pyhd8ed1ab_0    conda-forge
---
> markupsafe                2.1.2            py39ha30fb19_0    conda-forge
195,204c150
< mypy_extensions           0.4.3            py39h6e9494a_6    conda-forge
< mysql-common              8.0.31               hc4b2c72_0    conda-forge
< mysql-libs                8.0.31               h8658499_0    conda-forge
< nb_conda_kernels          2.3.1            py39h6e9494a_2    conda-forge
< nbclassic                 0.4.8              pyhd8ed1ab_0    conda-forge
< nbclient                  0.7.2              pyhd8ed1ab_0    conda-forge
< nbconvert                 7.2.7              pyhd8ed1ab_0    conda-forge
< nbconvert-core            7.2.7              pyhd8ed1ab_0    conda-forge
< nbconvert-pandoc          7.2.7              pyhd8ed1ab_0    conda-forge
< nbformat                  5.7.1              pyhd8ed1ab_0    conda-forge
---
> mypy_extensions           1.0.0              pyha770c72_0    conda-forge
206d151
< nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
209,213c154
< notebook                  6.5.2              pyha770c72_1    conda-forge
< notebook-shim             0.2.2              pyhd8ed1ab_0    conda-forge
< nspr                      4.35                 hea0b92c_0    conda-forge
< nss                       3.78                 ha8197d3_0    conda-forge
< numpy                     1.24.0           py39hdfa1d0c_0    conda-forge
---
> numpy                     1.24.2           py39h6ee2318_0    conda-forge
216c157
< packaging                 22.0               pyhd8ed1ab_0    conda-forge
---
> packaging                 23.0               pyhd8ed1ab_0    conda-forge
218,221c159
< pandoc                    2.19.2               h694c41f_1    conda-forge
< pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
< pango                     1.50.12              hbd9bf65_1    conda-forge
< parso                     0.8.3              pyhd8ed1ab_0    conda-forge
---
> pango                     1.50.13              hbd9bf65_0    conda-forge
224c162
< pathspec                  0.10.3             pyhd8ed1ab_0    conda-forge
---
> pathspec                  0.11.0             pyhd8ed1ab_0    conda-forge
227,228d164
< pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
< pickleshare               0.7.5                   py_1003    conda-forge
231,233c167,168
< pkginfo                   1.9.2              pyhd8ed1ab_0    conda-forge
< pkgutil-resolve-name      1.3.10             pyhd8ed1ab_0    conda-forge
< platformdirs              2.6.0              pyhd8ed1ab_0    conda-forge
---
> pkginfo                   1.9.6              pyhd8ed1ab_0    conda-forge
> platformdirs              3.0.0              pyhd8ed1ab_0    conda-forge
235d169
< ply                       3.11                       py_1    conda-forge
239d172
< prometheus_client         0.15.0             pyhd8ed1ab_0    conda-forge
243,244d175
< ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
< pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
247d177
< pycodestyle               2.10.0             pyhd8ed1ab_0    conda-forge
251d180
< pyflakes                  3.0.1              pyhd8ed1ab_0    conda-forge
253c182
< pygments                  2.13.0             pyhd8ed1ab_0    conda-forge
---
> pygments                  2.14.0             pyhd8ed1ab_0    conda-forge
257c186
< pyopenssl                 22.1.0             pyhd8ed1ab_0    conda-forge
---
> pyopenssl                 23.0.0             pyhd8ed1ab_0    conda-forge
259,261d187
< pyqt                      5.15.7           py39h327fbb6_2    conda-forge
< pyqt5-sip                 12.11.0          py39hc100f75_2    conda-forge
< pyrsistent                0.19.2           py39ha30fb19_0    conda-forge
268d193
< python-fastjsonschema     2.16.2             pyhd8ed1ab_0    conda-forge
270d194
< python-json-logger        2.0.4              pyhd8ed1ab_0    conda-forge
273c197
< python-utils              3.4.5              pyhd8ed1ab_0    conda-forge
---
> python-utils              3.5.2              pyhd8ed1ab_0    conda-forge
275c199
< pytz                      2022.7             pyhd8ed1ab_0    conda-forge
---
> pytz                      2022.7.1           pyhd8ed1ab_0    conda-forge
277,281d200
< pyzmq                     24.0.1           py39hed8f129_1    conda-forge
< qt-main                   5.15.6               haeff654_5    conda-forge
< qtconsole                 5.4.0              pyhd8ed1ab_0    conda-forge
< qtconsole-base            5.4.0              pyha770c72_0    conda-forge
< qtpy                      2.3.0              pyhd8ed1ab_0    conda-forge
297d215
< send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
299c217
< setuptools                67.4.0             pyhd8ed1ab_0    conda-forge
---
> setuptools                65.6.3             pyhd8ed1ab_0    conda-forge
302d219
< sip                       6.7.5            py39h7a8716b_0    conda-forge
305d221
< sniffio                   1.3.0              pyhd8ed1ab_0    conda-forge
310c226
< sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
---
> sphinxcontrib-applehelp   1.0.4              pyhd8ed1ab_0    conda-forge
312c228
< sphinxcontrib-htmlhelp    2.0.0              pyhd8ed1ab_0    conda-forge
---
> sphinxcontrib-htmlhelp    2.0.1              pyhd8ed1ab_0    conda-forge
316,317d231
< sqlite                    3.40.0               h9ae0607_0    conda-forge
< stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
324,325d237
< terminado                 0.17.1             pyhd1c38e8_0    conda-forge
< tinycss2                  1.2.1              pyhd8ed1ab_0    conda-forge
333d244
< traitlets                 5.8.0              pyhd8ed1ab_0    conda-forge
339,340c250,251
< urllib3                   1.26.13            pyhd8ed1ab_0    conda-forge
< virtualenv                20.17.1          py39h6e9494a_0    conda-forge
---
> urllib3                   1.26.14            pyhd8ed1ab_0    conda-forge
> virtualenv                20.19.0            pyhd8ed1ab_0    conda-forge
342,344c253
< wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
< webencodings              0.5.1                      py_1    conda-forge
< websocket-client          1.4.2              pyhd8ed1ab_0    conda-forge
---
> wcwidth                   0.2.6              pyhd8ed1ab_0    conda-forge
346,347c255
< widgetsnbextension        4.0.5              pyhd8ed1ab_0    conda-forge
< wrapt                     1.14.1           py39ha30fb19_1    conda-forge
---
> wrapt                     1.15.0           py39ha30fb19_0    conda-forge
353d260
< zeromq                    4.3.4                he49afe7_1    conda-forge
355c262
< zipp                      3.11.0             pyhd8ed1ab_0    conda-forge
---
> zipp                      3.15.0             pyhd8ed1ab_0    conda-forge
357,358c264
< zstandard                 0.19.0           py39h5bcbcff_1    conda-forge
< zstd                      1.5.2                hfa58983_4    conda-forge
---
> zstd                      1.5.2                hbc0c0cd_6    conda-forge

```

**Comment by ericdill** [comment]
depfinder does pull the name mapping from cf-graph-countyfair [at runtime](https://github.com/ericdill/depfinder/blob/83979857e119dff8474423d5514b00ce66c1f4e9/depfinder/utils.py#L71). is there a chance we're encountering some issue with that?

oh actually there's other files that get pulled in too:
- [import_maps_meta.json](https://github.com/ericdill/depfinder/blob/83979857e119dff8474423d5514b00ce66c1f4e9/depfinder/reports.py#L50)
- [import_maps/](https://github.com/ericdill/depfinder/blob/83979857e119dff8474423d5514b00ce66c1f4e9/depfinder/reports.py#L59)
- [ranked_hubs_authorities.json](https://github.com/ericdill/depfinder/blob/83979857e119dff8474423d5514b00ce66c1f4e9/depfinder/reports.py#L71)

I see that in this repo we're also pulling the [ranked_hubs_authorities](
https://github.com/regro/cf-scripts/pull/1619/files#diff-72c424d8b7636de572e6c843c301baf38b792417b2b0359c88a68a1e93e6d70bR238) file. maybe something to do with that? i'll poke a bit more

**Comment by ericdill** [comment]
some more debugging info. something to do with the depfinder comparison function in update_deps.py isn't operating properly. locally there's a bunch of additional files downloaded. Remotely nothing extra gets downloaded.

<details>
  <summary>logging from CI</summary>

```
Extracting download
DEBUG    depfinder:inspection.py:338 Setting PACKAGE_NAME global variable to work
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:456 [https://raw.githubusercontent.com:443](https://raw.githubusercontent.com/) "GET /regro/libcfgraph/master/.file_listing.json HTTP/1.1" 200 6473322
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:456 [https://raw.githubusercontent.com:443](https://raw.githubusercontent.com/) "GET /regro/cf-graph-countyfair/master/ranked_hubs_authorities.json HTTP/1.1" 200 110570
DEBUG    conda_forge_tick.update_deps:update_deps.py:246 deps from source: {'required': {}, 'questionable': {}, 'builtin': {}, 'questionable no match': {}, 'required no match': {}}
DEBUG    conda_forge_tick.update_deps:update_deps.py:253 depfinder audit: {}
INFO     conda_forge_tick.update_deps:update_deps.py:58 source dep. comp: {'run': {}}
{'run': {}} 

Dependency Analysis
--------------------
```

</details>

<details>
  <summary>locally</summary>

```
Extracting download
DEBUG    depfinder:inspection.py:342 Setting PACKAGE_NAME global variable to work
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/.file_listing.json HTTP/1.1" 200 6473322
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/cf-graph-countyfair/master/ranked_hubs_authorities.json HTTP/1.1" 200 110570
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps_meta.json HTTP/1.1" 200 40
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps_meta.json HTTP/1.1" 200 40
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps_meta.json HTTP/1.1" 200 40
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps_meta.json HTTP/1.1" 200 40
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps_meta.json HTTP/1.1" 200 40
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps_meta.json HTTP/1.1" 200 40
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps_meta.json HTTP/1.1" 200 40
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps_meta.json HTTP/1.1" 200 40
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps_meta.json HTTP/1.1" 200 40
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:1003 Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps/setup.json HTTP/1.1" 200 791111
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps/praw..json HTTP/1.1" 200 781
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps/webso.json HTTP/1.1" 200 13850
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps/reque.json HTTP/1.1" 200 5834
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps/reque.json HTTP/1.1" 200 5834
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps/praw.json HTTP/1.1" 200 134
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps/updat.json HTTP/1.1" 200 147
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps/prawc.json HTTP/1.1" 200 193
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://raw.githubusercontent.com:443 "GET /regro/libcfgraph/master/import_maps/prawc.json HTTP/1.1" 200 193
DEBUG    conda_forge_tick.update_deps:update_deps.py:246 deps from source: {'required': {'websocket': {'websocket-client'}, 'praw.endpoints': {'praw'}, 'prawcore': {'prawcore'}, 'setuptools': {'setuptools'}, 'requests.exceptions': {'arm_pyart', 'google-cloud-bigquery-storage-core', 'dbxfs', 'apache-libcloud', 'google-api-core', 'requests', 'autovizwidget'}, 'prawcore.exceptions': {'prawcore'}}, 'questionable': {'update_checker': {'update_checker'}, 'praw': {'praw'}, 'requests': {'arm_pyart', 'google-cloud-bigquery-storage-core', 'dbxfs', 'apache-libcloud', 'google-api-core', 'requests', 'autovizwidget'}}, 'builtin': {}, 'questionable no match': {}, 'required no match': {}}
DEBUG    conda_forge_tick.update_deps:update_deps.py:253 depfinder audit: {'df_minus_cf': {'prawcore', 'websocket-client', 'requests'}}
INFO     conda_forge_tick.update_deps:update_deps.py:58 source dep. comp: {'run': {'df_minus_cf': {'prawcore', 'websocket-client', 'requests'}}}
{'run': {'df_minus_cf': {'prawcore', 'websocket-client', 'requests'}}}

Dependency Analysis
--------------------
```

</details>

**Comment by ericdill** [comment]
okay ive added debug logs to as much of this project as possible. next steps are to add debug logging to depfinder itself. my approach for that will be to push a new depfinder branch with some debug logging and then pip install from that branch in this CI so that we can take advantage of it without needing to make a new depfinder release. i'll start working on that later this morning

**Comment by ericdill** [comment]
tracked it down: we're checking the full path of files against a list of fnmatch patterns. The default env name for the [setup-miniconda](https://github.com/conda-incubator/setup-miniconda) github action is `test`. this was triggering a match against the fnmatch patterns. The fix is to have depfinder not compare on the full path, but instead the current working dir. Then the conda env name is irrelevant.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -42,21 +43,43 @@ jobs:
...
+          echo "======================================================="
+          # pip uninstall depfinder -y
+          # mamba remove depfinder -y
+          # pip install -U git+https://github.com/ericdill/depfinder@debug-logging --no-deps
+          wget --quiet https://pypi.io/packages/source/p/praw/praw-7.7.0.tar.gz
+          tar xzf praw-7.7.0.tar.gz
+          python test_depfind.py
+          echo -e "\n======================================================="
+          echo "======================================================="
+          pytest -vvs tests -k test_get_dep_updates_and_hints_praw --showlocals --log-cli-level debug
```
Comment (beckermr) [inline .github/workflows/tests.yml:82]:
```suggestion
          pytest \
            -v \
            --cov=conda_forge_tick \
            --cov=tests \
            --cov-config=.coveragerc \
            --cov-report term \
            --durations 10 \
            tests
          codecov -X gcov
```

**Pull request merged**